lib/src/signed_video_h26x_auth.c

```diff
@@ -691,6 +691,52 @@ compute_gop_hash(signed_video_t *self, h26x_nalu_list_item_t *sei)
   return status;
 }
 
+/**
+ * Decodes the SEI message, retrieves necessary parameters for authentication, and computes the hash
+ * for authenticity.
+ */
+static svrc_t
+prepare_golden_sei(signed_video_t *self)
+{
+  assert(self);
+  h26x_nalu_list_t *nalu_list = self->nalu_list;
+  svrc_t status = SV_UNKNOWN_FAILURE;
+  sign_or_verify_data_t *verify_data = self->verify_data;
+  const size_t hash_size = verify_data->hash_size;
+
+  SV_TRY()
+    // Get the next SEI item from the NALU list.
+    h26x_nalu_list_item_t *sei = h26x_nalu_list_get_next_sei_item(nalu_list);
+
+    // If an SEI item is found and it hasn't been decoded yet, decode it.
+    if (sei && !sei->has_been_decoded) {
+      // Extract the TLV data and size from the NALU.
+      const uint8_t *tlv_data = sei->nalu->tlv_data;
+      size_t tlv_size = sei->nalu->tlv_size;
+
+      // Decode the SEI data and update the status.
+      SV_THROW(decode_sei_data(self, tlv_data, tlv_size));
+      sei->has_been_decoded = true;  // Mark the SEI as decoded.
+      if (self->gop_info->signature_hash_type == DOCUMENT_HASH) {
+        SV_THROW(hash_and_add_for_auth(self, sei));
+        memcpy(verify_data->hash, sei->hash, hash_size);
+      }
+    }
+
+    // Check if the SEI is decoded, not yet used in the GOP hash, and the signature hash type is
+    // GOP_HASH.
+    if (sei && sei->has_been_decoded && !sei->used_in_gop_hash &&
+        self->gop_info->signature_hash_type == GOP_HASH) {
+      // Compute the hash and add it for authentication.
+      SV_THROW(hash_and_add_for_auth(self, sei));
+    }
+    SV_THROW(prepare_for_validation(self));
```

Comment:
If the Golden SEI is not the first NALU of the stream, the signature will not be verified. This is what will happen if signing is done in a separate thread; See line 814.